1-exercises/B-highlight-words/script.js

```diff
@@ -1,5 +1,41 @@
 function highlightWords(paragraph, colours) {
   // Write your code here...
+
+  const para = document.createElement("p");
+  const selectColour = document.createElement("select");
+
+  document.querySelector("#content").appendChild(para);
+  document.querySelector("#content").appendChild(selectColour);
+
+  ["Please choose a colour", ...colours].forEach((colour) => {
+    let colourOption = document.createElement("option");
+    colourOption.value = colour;
+    colourOption.innerText = colour[0].toUpperCase() + colour.slice(1);
+    selectColour.appendChild(colourOption);
+  });
+
+  paragraph.split(" ").forEach((word) => {
+    const span = document.createElement("span");
+    span.innerText = word + " ";
+    para.appendChild(span);
+
+    span.addEventListener("click", function () {
+      switch (document.querySelector("select").value) {
+        case "yellow":
+          span.style.backgroundColor = "yellow";
+          break;
+        case "green":
+          span.style.backgroundColor = "green";
+          break;
+        case "blue":
+          span.style.backgroundColor = "blue";
+          break;
+        default:
+          span.style.backgroundColor = "none";
+          break;
+      }
+    });
+  });
 }
```

Comment:
Nice one! I really liked that you used switch! well done
I would only suggest creating CSS classes and manipulating them, which is generally more efficient because it allows the browser to use its built-in CSS rendering engine to update the styles. But, on the other hand it could be slower because it requires updating the DOM tree for each span element.
So, this is just personal preference!